homeassistant/components/mqtt/subscription.py

```diff
@@ -0,0 +1,51 @@
+"""
+Helper to handle a set of topics to subscribe to.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/mqtt/
+"""
+import logging
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import DEFAULT_QOS, MessageCallbackType
+from homeassistant.loader import bind_hass
+from homeassistant.helpers.typing import (
+    HomeAssistantType)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@bind_hass
+async def async_subscribe_topics(hass: HomeAssistantType, sub_state: dict,
```

Comment:
This helper should get some tests added, including:
 - updating a topic QOS/encoding
 - removing a topic in 2nd call
 - adding a topic in 2nd call
 - passing in same data should not do anything

Reply:
And once it's tested here, for all other platforms using this function we just test that it's called with the right params 👍

Reply:
Tests added.